finish/src/main/java/io/openliberty/guides/microprofile/InventoryResource.java

```diff
@@ -53,9 +55,24 @@ public JsonObject listContents() {
     if (!inventoryConfig.isInMaintenance()) {
       return manager.list();
     } else {
-      return JsonMessages.serviceInMaintenance("InventoryResource");
+      return returnMessage();
     }
   }
   // end::config-methods[]
 
+  // tag::returnMessage[]
+  public JsonObject returnMessage() {
+    // A use case of custom converter for Email class type
+    Email devEmail = inventoryConfig.getEmail();
+    JsonObject contact = Json.createObjectBuilder()
+                             .add("Email", devEmail.toString())
+                             .add("Name", devEmail.getEmailName())
+                             .add("Domain", devEmail.getEmailDomain()).build();
+    return Json.createObjectBuilder()
+               .add("Message",
+                    JsonMessages.serviceInMaintenance("InventoryResource"))
+               .add("Contact", contact).build();
+  }
+  // end::returnMessage[]
+
```

Comment:
This can be done in JsonMessage still?

Reply:
JsonMessage must be static, however, the email is a non-static object, so we cannot put it in the JsonMessage class.